napari/plugins/_npe2.py

```diff
@@ -501,18 +514,44 @@ def _register_manifest_actions(mf: PluginManifest) -> None:
     app = get_app()
     actions, submenus = _npe2_manifest_to_actions(mf)
     samples_submenu, sample_actions = _get_samples_submenu_actions(mf)
+    widgets_submenu: List[Any] = []
+    widget_actions: List[Any] = []
+    try:
+        from napari._qt._qplugins import _get_widgets_submenu_actions
+    except ModuleNotFoundError:
+        pass
+    else:
+        widgets_submenu, widget_actions = _get_widgets_submenu_actions(mf)
```

Comment:
Yeah it totally is! I even thought about moving both these things to a qt place, and then do something like:

```
    for mf in manifests:
        if not pm.is_disabled(mf.name):
            _register_manifest_actions(mf)
            _register_widgets(mf)
```

And now I think about it more, it's may be easier than I originally though. Maybe this would be the cleaner option?

Reply:
ALso @DragaDoncila WDYT?

Reply:
Okay I've moved it but I am not sure where would be best to put the qt guard. I'm not sure if I can put it in `_qnpe2.py` (as its the `__init__.py` in `qt/` that causes the problem) so it's in `_npe2.py`. Happy to change.

Reply:
Hmm yeah I agree it's a little jarring to see the `qt` checks there, but also I think you're right that if you put it in `_qnpe2` we'll be back to square one. I think we should at least put the try/except block into its own function in `_npe2` so that we don't have to repeat it both in `on_plugins_registered` and `on_plugin_enablement_change`